lib/vrl/parser/src/ast.rs

```diff
@@ -742,13 +747,36 @@ pub enum Assignment {
     // }
 }
 
+#[derive(Clone, PartialEq)]
+pub enum AssignmentOp {
+    Assign,
+    Merge,
+}
+
+impl fmt::Display for AssignmentOp {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use AssignmentOp::*;
+
+        match self {
+            Assign => write!(f, "="),
+            Merge => write!(f, "|="),
+        }
+    }
+}
+
+impl fmt::Debug for AssignmentOp {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::Display::fmt(self, f)
```

Comment:
Most (all?) debug impls in the AST do something like `AssignmentOp(=)` and `AssigmentOp(|=)`.

We didn't for the assignment operator before because you could only use `=`, but now that's changed, it'd be nice to stick to that convention.